### ISSUE TITLE
Theme

### ISSUE THREAD

**Issue by mottosso** [mottosso/allzpark#14]
### Goal

Add at least one additional theme, a dark one.

### Motivation

The most requested feature has been the ability to change the theme. Possibly because it's the first thing you see as you enter the app and therefore the thing that sticks with you during your journey.

### Implementation

Use the #13 to craft a dark theme, save it to file and offer it as an option in the Preferences Tab.

**Comment by davidlatwe** [comment]
So this is my current dark themed Allzpark (on top of #110) :

![image](https://user-images.githubusercontent.com/3357009/96288164-8d8b3400-1015-11eb-9839-8f7467ebb28c.png)

The dark style theme was forked from Avalon, which I think it was originally shipped by @BigRoy. 
And it's gorgeous.

Here's how I implemented :
1. Allzpark firstly loads it's own style
2. Then loads Avalon dark style theme
3. Lastly, a few style changes which are specifically to override Allzpark's style, e.g. `Window {background: #242424;}` (note that the `Window` is the widget that named in Allzpark)

As you can see, there are lots of style-sheet properties overriding each other, and not easy to manage or merge.
And maybe it could also be really hard for user to made their own change ?

So I guess if we want to ship a default dark style theme, Avalon's dark theme is not an option ?

**Comment by davidlatwe** [comment]
I have re-crafted two default themes "light" and "dark", and can be switched in preference panel !

![5ed3f312-aa60-4f0b-951a-feeb5db2a756](https://user-images.githubusercontent.com/3357009/96385232-214f3280-11c5-11eb-8d8e-f27c51ff1587.gif)

This implementation (draft) is currently on my branch [styling-2](https://github.com/davidlatwe/allzpark/tree/styling-2).

And user can make their own style color *palette* in `allzparkconfig.py` as their custom themes, for example :

```python
# allzparkconfig.py
palettes = {
    # These will be picked up in preference
    "another_dark_theme": {
        "brightest": "#403E3D",
        "bright": "#383635",
        "base": "#2E2C2C",
        "dim": "#21201F",
        "dimmest": "#141413",

        "highlight": "#69D6C2",
        "highlighted": "#111111",
        "active": "silver",
        "inactive": "dimGray",
    },
}
```

#### Caveat
While I was testing on PySide2 (Qt version 5.13.0), switching theme stylesheet is buggy (style not all applied), restart Allzpark is required.

But PySide2 (Qt version 5.15.1) does not have this issue. As the above GIF shown.

**Comment by mottosso** [comment]
>So I guess if we want to ship a default dark style theme, Avalon's dark theme is not an option ?

I don't see that as a problem. Even without a stylesheet, every widget would have a style you'd need to override. The Avalon CSS just changes that style, and the user would have to change it either way. Or am I misunderstanding your concern?

> I have re-crafted two default themes "light" and "dark", and can be switched in preference panel !

That is hooot! :D

> While I was testing on PySide2 (Qt version 5.13.0), switching theme stylesheet is buggy (style not all applied), restart Allzpark is required.

That's fine to me, maybe just put a notice in the README. But having to restart isn't a showstopper and is probably somewhat expected anyway.

**Comment by davidlatwe** [comment]
> Even without a stylesheet, every widget would have a style you'd need to override. The Avalon CSS just changes that style, and the user would have to change it either way. Or am I misunderstanding your concern?

No, you aren't. But Avalon CSS is just too many lines, and it's really not easy to make a light-color version from it XD.
So I end up re-crafting them from ground up, with minimum colors and additional icons.

BTW, since this switchable default themes is on top of #110 in my end, do you mind if I continue push it into that PR ? Or you prefer a different one after #110 is merged ?

**Comment by mottosso** [comment]
> But Avalon CSS is just too many lines, and it's really not easy to make a light-color version from it XD.

Oh, ok. But, the version of Allzpark *without* this theme is already light? Could we use the new Dark theme a default, and then just remove it when chosing the Light mode?

**Comment by mottosso** [comment]
> Or you prefer a different one after #110 is merged ?

I have no preference, whichever way is easier for you to manage!

**Comment by davidlatwe** [comment]
> Could we use the new Dark theme a default, and then just remove it when chosing the Light mode?

I had thought about that, but then I thought it might be much better if we ship only one style-sheet that is string-formatable, e.g. :

```css
QPushButton {
    border: 1px solid %(dim)s;
    background: %(brightest)s;
}
```

 and user only need to provide their own color-tickets to fill up that style-sheet for changing color quickly.

And one other benefit from this is, that style-sheet is just enough to affect the widgets that are being used in Allzpark, no much no less.

So I think this would be a good setup for default.

**Comment by davidlatwe** [comment]
Implemented in #111
